frontend_csv/src/scanner/scanner.rs

```diff
@@ -1,13 +1,160 @@
 use super::token::CSVToken;
-
+use super::token::CSVTokenType;
+use std::collections::VecDeque;
 /// Core implementation of the scanner
 pub struct Scanner {
-    pos: usize,
-    line: u64,
-    tokens: Vec<CSVToken>,
-    identifier_symbols: Vec<char>,
+    pub pos: usize,
+    // line: u64,
+    // tokens: Vec<CSVToken>,
+    // identifier_symbols: Vec<char>,
 }
 
 impl Scanner {
-    fn scan(&mut self, _symbols: &Vec<char>) {}
+    pub fn scan(&mut self, _symbols: &Vec<char>) -> VecDeque<CSVToken> {
+        let mut tokenize: VecDeque<CSVToken> = VecDeque::new();
+        let mut currentBuffer: String = String::new();
```

Comment:
```suggestion
        let mut current_buffer: String = String::new();
```

Rust here will complain during compilation, so you can catch this at compile time